pkgs/tools/networking/knock/package.nix

```
@@ -0,0 +1,31 @@
+{ lib, buildGoModule, installShellFiles, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "knock";
+  version = "0.0.2";
+
+  src = fetchFromGitHub {
+    owner = "nat-418";
+    repo = pname;
+    rev = "a3749685381cae178bb5836c67645e0fce7aa1d0";
```

The source is [tagged](https://github.com/nat-418/knock/releases/tag/v0.0.2).

```suggestion
    rev = "refs/tags/v${version}";
```